src/modules/tradeQuote/hooks/useQuoteParams.ts

```diff
@@ -1,18 +1,33 @@
+import { useAtomValue } from 'jotai'
 import { useMemo } from 'react'
 
 import { OrderKind } from '@cowprotocol/cow-sdk'
 
 import { useDerivedTradeState } from 'modules/trade/hooks/useDerivedTradeState'
+import { TradeType, useTradeTypeInfo } from 'modules/trade/hooks/useTradeTypeInfo'
+import { partsStateAtom } from 'modules/twap/state/partsStateAtom'
 import { useWalletInfo } from 'modules/wallet'
 
 import { getAddress } from 'utils/getAddress'
 
 export function useQuoteParams() {
+  const tradeTypeInfo = useTradeTypeInfo()
+
   const { chainId, account } = useWalletInfo()
   const { state } = useDerivedTradeState()
+  const { inputPartAmount } = useAtomValue(partsStateAtom)
```

Comment:
We can't use `modules/twap` in `modules/trade`, it's violation of Dependency inversion principle

Reply:
We need `inputPartAmount` here, any suggestion?

Reply:
https://github.com/cowprotocol/cowswap/pull/2586